src/otx/core/data/dataset/detection.py

```diff
@@ -49,11 +49,20 @@ def _get_item_impl(self, index: int) -> DetDataEntity | None:
                 bboxes,
                 format=tv_tensors.BoundingBoxFormat.XYXY,
                 canvas_size=img_shape,
-            ),
+            )
+            if self.dm_subset.name == "train"
+            else torch.as_tensor(bboxes),  # to avoid resizing bounding boxes when validation and testing
```

Comment:
You are creating a if-else branch in the common module. However, it is extremely dedicated to just one local case where a data pipeline produces resized bounding box, but a model does not allow it. This is basically not desired and turning back the clock to make OTX 2 into OTX 1.

Reply:
I agree with your comment, but this is needed if using `torchvision.transforms.v2.Resize` because it forces to transform tensors if there is `BoundingBoxes` tv_tensor.
I'm seeking the appropriate location for this patching and I'll update it.

Reply:
We don't use torchvision operations in this PR, so we can handle whether transforming bboxes or not in the data pipeline, not common modules.
And reverting transformed bboxes in metric side has two hurdles: it is needed to add additional things into `_convert_pred_entity_to_compute_metric` in `src/otx/core/model/detection.py`, and it might be complicated to use model's postprocessing functions.
So, I added a flag about this function at https://github.com/openvinotoolkit/training_extensions/pull/3253/commits/c9bcb3673721eaa63cbbc4ed09d45811f8cc2fc8 because it is the simplest way of this issue.
If you think it should be done on metric side, not data pipeline side, I'll consider to update it when merging model and data sides.